scripts/features.R

```diff
@@ -0,0 +1,125 @@
+
+
+
+# read all the forms into a list of dataframes using colwise to guess the column types
+# if we don't try to guess the col types we have issues later with the bind_rows join
+form <- form_names_l %>%
+  purrr::map(sf::st_read) %>%
+  purrr::map(plyr::colwise(type.convert)) %>%
+  # name the data.frames so we can add it later as a "source" column - we use basename to leave the filepath behind
+  purrr::set_names(nm = basename(form_names_l)) %>%
+  bind_rows(.id = 'source')
+
+
+# see the names of our form
+names(form)
+
+# select the columns we want
+
+features <- form %>%
+  select(local_name, feature_type, feature_height_m, feature_length_m, feature_height_length_method, feature_time) %>%
+  filter(!is.na(feature_type))
+
+features2 <- form %>%
+  select(local_name, feature_type_2, feature_height_2_m, feature_length_2_m, feature_height_length_method, feature_time_2) %>%
+  filter(!is.na(feature_type_2)) %>%
+  rename(feature_type = feature_type_2,
+         feature_height_m = feature_height_2_m,
+         feature_length_m = feature_length_2_m,
+         feature_time = feature_time_2) %>%
+  # there was bug in the mergin form where debris jam couldn't be added as feature type so it was included in length column
+  mutate(feature_type = case_when(feature_length_m == '2m debris jam' ~ 'LWD Jam', T ~ feature_type)) %>%
+  mutate(feature_length_m = case_when(feature_length_m == '2m debris jam' ~ '2', T ~ feature_length_m)) %>%
+  mutate(feature_length_m = as.numeric(feature_length_m))
+
+features3 <- form %>%
+  select(local_name, feature_type_3, feature_height_3_m, feature_length_3_m, feature_height_length_method, feature_time_3) %>%
+  filter(!is.na(feature_type_3)) %>%
+  rename(feature_type = feature_type_3,
+         feature_height_m = feature_height_3_m,
+         feature_length_m = feature_length_3_m,
+         feature_time = feature_time_3) %>%
+  # there was bug in the mergin form where debris jam couldn't be added as feature type so it was included in height column
+  mutate(feature_type = case_when(feature_height_m == '2 log jam' ~ 'LWD Jam', T ~ feature_type)) %>%
+  mutate(feature_height_m = case_when(feature_height_m == '2 log jam' ~ '2', T ~ feature_height_m)) %>%
+  mutate(feature_height_m = as.numeric(feature_height_m))
+
+features_final <- bind_rows(features, features2, features3) %>%
+  mutate(feature_time = lubridate::as_datetime(feature_time, tz="America/Vancouver"))
+
+# get UTMS of features that have times by joining to tracks
+# read in tracks
+track_points_prep = read_sf('data/habitat_confirmation_tracks.gpx', layer = "track_points")
+
+track_points <- track_points_prep %>%
+  st_coordinates() %>%
+  as_tibble() %>%
+  setNames(c("gps_latitude","gps_longitude")) %>%
+  rowid_to_column()
+
+#find the track point that is closest in time to the CreateDate stamp of the photo..
+#https://stackoverflow.com/questions/21607247/searching-for-nearest-date-in-data-frame
+get_closest_line_in_history <- function(x, history){
```

Comment:
we should be finishing the script where this was done first before trying to leverage it forward. that allows us to not duplicated effort and/or issues